### PR TITLE
feat(LAB-4291): add copy_workflow_from_project SDK method

### PR DESCRIPTION
**Pull request by RuellePaul**
Add a high-level method to copy a workflow from one project to another, including step configurations (consensus, coverage, sendBackStepId).

- Add `copy_workflow_from_project` to presentation client and use cases
- Add `send_back_step_id` field to WorkflowStepCreate and WorkflowStepUpdate
- Add `sendBackStepId` mapping in the GraphQL step mapper
- Validate destination project has no labels before copying
- Remap sendBackStepId references to new step IDs after creation
- Don't copy assignees per specification
- Add unit tests for all scenarios

### CODE REVIEW

**Review by baptiste-olivier — DISMISSED** [review]
Can you add a notebook, and add it to our e2e stack ? 🙏

**Review by FannyGaudin — APPROVED** [review]

**Review by aurelienlombard — APPROVED** [review]

**Review by baptiste-olivier — COMMENTED** [review]
## Auto Review Summary

**Verdict:** Needs discussion (already merged)

| Severity | Count |
|----------|-------|
| Blocker | 0 |
| Required | 1 |
| Suggestion | 3 |
| Question | 2 |
| Praise | 2 |

Well-structured feature with good validations and test coverage, but one potential bug in the first-step update and some architectural concerns.

---

### [PRAISE] Comprehensive validations

Good defensive checks: same project ID guard, destination has no labels, duplicate step names detection, and labeler count for consensus. These prevent confusing runtime errors.

### [PRAISE] Thorough test coverage

11 test cases covering happy path, error conditions, sendBackStepId remapping, assignee routing, and edge cases (no existing workflow, extra dest steps). Well-structured test helpers (`_make_source_steps`, `_setup_happy_path`).

File: src/kili/use_cases/project_workflow/__init__.py [hunk 1]
```diff
@@ -50,3 +64,280 @@ def remove_reviewers_from_step(
...
+            for step in source_steps:
+                assignees = labeler_ids if step.get("type") == "DEFAULT" else reviewer_ids
+                steps_to_create.append(_make_create_step(step, assignees))
+                if step.get("sendBackStepId"):
+                    source_steps_with_send_back.append(
+                        (str(step["name"]), str(step["sendBackStepId"]))
+                    )
+
+        delete_steps = [str(step["id"]) for step in dest_steps[1:]] or None
+
```
Comment (baptiste-olivier) [inline src/kili/use_cases/project_workflow/__init__.py:155]: [REQUIRED] `_build_step_update` doesn't copy `stepCoverage` or `type` for the first step

When updating the first destination step to match the first source step, only `name`, `consensusCoverage`, and `numberOfExpectedLabelsForConsensus` are set. The `stepCoverage` and `type` fields from the source step are **not** copied. If the first source step has a `stepCoverage` value, or a different `type` than the existing destination step, those properties will be silently lost.

```python
# Current - missing stepCoverage and type
def _build_step_update(dest_step_id, source_step):
    update = {"id": dest_step_id, "name": ...}
    # only sets consensusCoverage, numberOfExpectedLabelsForConsensus
    # Missing: stepCoverage, type
    return update
```

File: src/kili/adapters/kili_api_gateway/project_workflow/types.py [hunk 2]
```diff
@@ -14,3 +14,4 @@ class ProjectWorkflowDataKiliAPIGatewayInput:
     create_steps: Optional[list[WorkflowStepCreate]]
     update_steps: Optional[list[WorkflowStepUpdate]]
     delete_steps: Optional[list[str]]
+    for_copy: bool = False
```
Comment (baptiste-olivier) [inline src/kili/adapters/kili_api_gateway/project_workflow/types.py:17]: [SUGGESTION] `for_copy` flag leaks through the architecture

This flag is threaded from the use case through `ProjectWorkflowDataKiliAPIGatewayInput` into the mapper, just to change serialization behavior (keeping `None` for consensus fields). The adapter layer shouldn't know about "copy" semantics. A cleaner approach would be to explicitly set `consensusCoverage=0` or have the use case pass a sentinel to signal "include this field even if None", keeping the mapper stateless.

File: src/kili/adapters/kili_api_gateway/project_workflow/mappers.py [hunk 3]
```diff
@@ -12,28 +10,39 @@ def project_input_mapper(data: ProjectWorkflowDataKiliAPIGatewayInput) -> dict:
...
+            ]
             if data.update_steps
             else [],
             "deletes": data.delete_steps if data.delete_steps else [],
         },
     }
 
 
-def update_step_mapper(data: Union[WorkflowStepCreate, WorkflowStepUpdate]) -> dict:
+def update_step_mapper(
```
Comment (baptiste-olivier) [inline src/kili/adapters/kili_api_gateway/project_workflow/mappers.py:28]: [SUGGESTION] Typo: "worklow" → "workflow". Also `##` is unusual for a Python comment — typically single `#`.

File: recipes/copy_workflow_between_projects.ipynb [hunk 4]
```diff
@@ -0,0 +1,421 @@
...
+     "output_type": "stream",
+     "text": [
+      "Source workflow steps:\n",
+      " - {'assignees': [{'email': 'test+edouard@kili-technology.com', 'id': 'user-2'}, {'email': 'labeler1@example.com', 'id': 'cmn8yvbve000lme4i2yxa252z'}, {'email': 'labeler2@example.com', 'id': 'cmn8yvbzi000tme4ifga589w7'}, {'email': 'reviewer1@example.com', 'id': 'cmn8yvc2h0011me4iajuh8v1r'}], 'type': 'DEFAULT', 'name': 'Labeling Renamed', 'id': 'cmn8z1apd004tme4i9bvbfwrv'}\n",
+      " - {'assignees': [{'email': 'test+edouard@kili-technology.com', 'id': 'user-2'}, {'email': 'reviewer1@example.com', 'id': 'cmn8yvc2h0011me4iajuh8v1r'}], 'type': 'REVIEW', 'name': 'Review', 'id': 'cmn8z1apn004ume4i8hwm0kce'}\n",
+      " - {'assignees': [{'email': 'test+edouard@kili-technology.com', 'id': 'user-2'}, {'email': 'test+max@kili-technology.com', 'id': 'user-7'}, {'email': 'test+queues@kili-technology.com', 'id': 'user-9'}, {'email': 'test+github@kili-technology.com', 'id': 'user-6'}, {'email': 'reviewer1@example.com', 'id': 'cmn8yvc2h0011me4iajuh8v1r'}], 'type': 'REVIEW', 'name': 'Review 2', 'id': 'cmn8z1e77005tme4ia3r05pk2'}\n"
+     ]
+    }
+   ],
+   "source": [
```
Comment (baptiste-olivier) [inline recipes/copy_workflow_between_projects.ipynb:183]: [SUGGESTION] Notebook outputs contain real-looking test environment emails (`test+edouard@kili-technology.com`, `test+max@kili-technology.com`) and internal IDs. Consider clearing outputs or using placeholder data to avoid leaking internal information.

File: src/kili/use_cases/project_workflow/__init__.py [hunk 5]
```diff
@@ -50,3 +64,280 @@ def remove_reviewers_from_step(
...
+        if source_project_id == destination_project_id:
+            raise ValueError(
+                "Source and destination project IDs must be different."
+                f" Got the same ID: {source_project_id}"
+            )
+
+        # 1. Fetch source workflow steps and settings
+        logger.info("Fetching workflow steps from source project %s", source_project_id)
+        source_steps = self._kili_api_gateway.get_steps(source_project_id, _SOURCE_STEP_FIELDS)
+
```
Comment (baptiste-olivier) [inline src/kili/use_cases/project_workflow/__init__.py:88]: [QUESTION] Consensus validation only checks the first step

`_validate_consensus_labelers` is only called for `source_steps[0]`. Is it guaranteed that only the first step can have consensus settings? If a later step could also have `numberOfExpectedLabelsForConsensus`, the copy would succeed but the destination workflow might be misconfigured.

File: src/kili/use_cases/project_workflow/__init__.py [hunk 6]
```diff
@@ -50,3 +64,280 @@ def remove_reviewers_from_step(
...
+    ) -> dict[str, object]:
+        """Copy a workflow from one project to another.
+
+        Fetches the source workflow steps, validates the destination project has no labels,
+        deletes existing destination steps, and creates new steps with remapped sendBackStepId
+        references.
+        """
+        if source_project_id == destination_project_id:
+            raise ValueError(
+                "Source and destination project IDs must be different."
```
Comment (baptiste-olivier) [inline src/kili/use_cases/project_workflow/__init__.py:81]: [QUESTION] No source project workflow V2 validation

The code validates the **destination** project is V2 but does not check the **source** project. If the source uses V1, `get_steps` might return unexpected data. Is this intentional because V1 projects can't have multi-step workflows anyway?
